### ISSUE TITLE
NYP subscription switching and resubscription

### ISSUE THREAD

**Issue by denzel100** [helgatheviking/woocommerce-nyp-subscription-switching#1]
If a subscription product is limited to "one subscription of any status" and a customer cancels that subscription, the product page contains an NYP field as if meaning to switch to a different price, but when checking out the customer gets redirected to the same product page containing only a resubscribe button that will allow the customer to resubscribe at the originally named price instead of the newly entered one. Once this page has been shown once, the NYP field apparently can only be made to re-appear by clicking the resubscribe button, which brings up the checkout screen, and then opening the product page again.

So basically, what appears to be going on is that the switching addon wants to switch and the Subs resubscription feature wants to to its thing instead. The clean solution would be to implement an NYP-on-resubscription feature. My current workaround is to limit the product to "one active subscription" instead of "one of any status". Visiting the product page then allows to create a new NYP subscription instead of resubscribing to the old one. The downside of this is that customers do no longer have a single, unique subscription that can be cancelled, renewed and switched.
